test/functional/mongo_client_tests.js

```diff
@@ -629,7 +629,7 @@ describe('MongoClient', function() {
     test: function(done) {
       var configuration = this.configuration;
       var url = configuration.url();
-      if (url.indexOf('replicaSet') !== -1) {
+      if (url.indexOf('rs_name') !== -1) {
```

Comment:
how do we know that our url will always have `rs_name` and not `replicaSet`? Could we check for both here?

More importantly, it looks like this is specifically trying to figure out whether to append a `?` or a `&` to the url string. Could we do:

```js
if (url.indexOf('?') !== -1) {
  url = `${configuration.url}&appname=hello%20world`;
} else {
  url = `${configuration.url}?appname=hello%20world`;
}
```

Reply:
yeah, actually this caused an error. For now I've reverted to `rs_name`, and will revert that to `replicaSet` in the new test runner branch. This is specific to how we build urls in the configuration, where the new config will build more up-to-date connection strings.